torch/distributions/transforms.py

```diff
@@ -412,6 +412,29 @@ def _inverse(self, y):
         return probs.log()
 
 
+class InvertableBoltzmannTransform(Transform):
+    """
+    Transform from unconstrained space to the simplex via `y = [exp(x), 1]` and
+    then normalizing. Note that dimensionality of `y.dim() == x.dim() + 1`.
+    """
+    domain = constraints.real
+    codomain = constraints.simplex
+    event_dim = 1
+
+    def __eq__(self, other):
+        return isinstance(other, InvertableBoltzmannTransform)
+
+    def _call(self, x):
+        ones = torch.ones(*(x.size()[:-1] + (1, )))
```

Comment:
This won't be on the correct device. Not sure what's the best one liner we could use right now, I'll need to think

Reply:
How about `probs = F.pad(x.exp(), (0, 1), value=1.)` instead of lines 428-29?